test/libsolidity/semanticTests/types/type_conversion_cleanup.sol

```diff
@@ -1,7 +1,7 @@
 contract Test {
-    function test() public returns (uint ret) { return uint(address(Test(address(0x11223344556677889900112233445566778899001122)))); }
+    function test() public returns (uint ret) { return uint(address(uint128(0x11223344556677889900112233445566778899001122))); }
```

Comment:
Is the intermediate conversion needed with the current code?

Reply:
I'm not if the change makes the test not valid anymore. What do you think?

Reply:
I don't understand your question.

Reply:
Sorry there was a typo before. I meant to ask if the change makes the test obsolete?

Reply:
Actually I'm wondering if we should also disallow the conversion from uint128 to address. I think the test is still valid.

Reply:
Do you mean any conversion between uint128 and address? Or Is it because of the narrowing conversion first? Related issue: https://github.com/ethereum/solidity/issues/9939

Reply:
this should be another PR, but yes: We should only convert one of "signedness", width and "kind" at the same time (kind being bytesNN, uint, address)